clang/include/clang-c/CAS.h

```diff
@@ -118,6 +119,41 @@ clang_experimental_cas_Databases_create(CXCASOptions Opts, CXString *Error);
  */
 CINDEX_LINKAGE void clang_experimental_cas_Databases_dispose(CXCASDatabases);
 
+/**
+ * Get the local storage size of the CAS/cache data in bytes.
+ *
+ * \param[out] OutError The error object to pass back to client (if any).
+ * If non-null the object must be disposed using \c clang_Error_dispose.
+ * \returns the local storage size of the CAS/cache data, or -1 if the
+ * implementation does not support reporting such size, or -2 if an error
+ * occurred.
+ */
+CINDEX_LINKAGE int64_t clang_experimental_cas_Databases_get_storage_size(
+    CXCASDatabases, CXError *OutError);
+
+/**
+ * Set the size for limiting disk storage growth.
+ *
+ * \param size_limit the maximum size limit in bytes. 0 means no limit.
+ * \param[out] OutError The error object to pass back to client (if any).
+ * If non-null the object must be disposed using \c clang_Error_dispose.
+ * \returns true if there was an error, false otherwise.
+ */
+CINDEX_LINKAGE bool clang_experimental_cas_Databases_set_size_limit(
+    CXCASDatabases, uint64_t size_limit, CXError *OutError);
```

Comment:
Maybe `CXErrorCode` instead of `bool` so the return values are named?  Or even return the `CXError` directly?  Same for other bool API below.

We should also consider limiting size to `int64_t`, since that's the maximum that `get_storage_size` can report.

Reply:
See https://github.com/apple/llvm-project/pull/7987/commits/654f789f94a84175d5bc3e1ac7444257c57f2267